File: backend/debug_mongo.py
import os
from pymongo import MongoClient
from dotenv import load_dotenv

load_dotenv()

uri = os.getenv('MONGO_URI')
client = MongoClient(uri)
db_name = uri.split('/')[-1].split('?')[0] or 'unirank'
db = client[db_name]

print(f"Connected to database: {db_name}")
collections = db.list_collection_names()
print(f"Collections in '{db_name}': {collections}")

for coll_name in collections:
    count = db[coll_name].count_documents({})
    print(f"Collection '{coll_name}': {count} documents")

File: backend/src/scripts/debugMongo.js
/**
 * scripts/debugMongo.js
 * Lists collections and document counts in MongoDB.
 */

import dotenv from 'dotenv';
import mongoose from 'mongoose';

dotenv.config();

const uri = process.env.MONGO_URI || 'mongodb://localhost:27017/unirank';

try {
  await mongoose.connect(uri);
  const db = mongoose.connection.db;
  console.log(`Connected to database: ${db.databaseName}`);
  const collections = await db.listCollections().toArray();
  console.log(`Collections in '${db.databaseName}':`, collections.map((c) => c.name));

  for (const coll of collections) {
    const count = await db.collection(coll.name).countDocuments();
    console.log(`Collection '${coll.name}': ${count} documents`);
  }
  await mongoose.disconnect();
} catch (err) {
  console.error('Debug mongo error:', err.message);
}
